staging/src/k8s.io/apiserver/pkg/storage/util.go

```diff
@@ -149,3 +149,30 @@ func HasInitialEventsEndBookmarkAnnotation(obj runtime.Object) (bool, error) {
 	objAnnotations := objMeta.GetAnnotations()
 	return objAnnotations[metav1.InitialEventsAnnotationKey] == "true", nil
 }
+
+// PrepareListRequestContinuationParametersIfApply prepares optional
+// parameters for retrieving additional results for a paginated request.
+//
+// This function sets up parameters that a client can use to fetch the remaining results
+// from the server if they are available.
+func PrepareListRequestContinuationParametersIfApply(keyLastItem, keyPrefix string, resourceVersion int64, itemsCount int64, hasMoreItems bool, opts ListOptions) (*int64, string, error) {
```

Comment:
Let's just call it `PrepareContinue` (or `PrepareContinueToken`).

Reply:
nit: and let's swap the order of result: `(string, *int64, error)`.
Also why *int64, not just int64 ?

Reply:
we need `*int64` because `remainingItemCount` will be used in (`nil` vs `0`)

https://github.com/kubernetes/kubernetes/blob/70e65eee46cbf72efabe8440a92117d1fedf6497/staging/src/k8s.io/apiserver/pkg/storage/interfaces.go#L55